Add updated publish documentation section (#121)
bump-sh/docs: src/_help/publish-documentation/deploy-and-release-management.md

---
title: Deploy and release management
---

- TOC
{:toc}

## What are deployments

> Deployment corresponds to the publication of an uploaded version of documentation.

Reviewer: Should we write "The deployment term" or "A deployment"?
Please remove the `version` term here.
We should also define what a release is, here.

When we talk about deployment, we mean the act of publishing a version of your documentation to your API users. A deployed version is a published, live version.

![](/images/help/deployments-list.png)

> By default, a version is automatically deployed after being uploaded. However, you can choose to deploy a version manually at a later time by using the Manual Release mode.

Reviewer: Just to add the idea, please rephrase if needed:
```suggestion
By default, a version is automatically deployed after being uploaded. However, you can choose to deploy a version manually at a later time by using the Manual Release mode. In this case, the latest deployments will be enqueued until you decide to release one of them.
```

> You can also add a title and description to your last automatic deployment afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.

Reviewer: ```suggestion
You can also add context (title and description) to any release afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.
```

### Deploy from the dashboard

To deploy a new version of documentation from the dashboard, you first need to go to the settings of that documentation.

From there, you will find the "Deploy new version" button, which will allow you to choose the file to upload and, if the documentation has branches, the branch to deploy it on.

![](/images/help/upload-version-quick.png)

After processing, the new version will be automatically deployed after a few moments unless you have enabled Manual Release mode.

> ### Deploy from the CLI

Reviewer: I'm in favor of replacing CLI/CI/GitHub action paragraphs with links pointing to the right section.

Author: Last Monday, we decided during the GTM weekly to share this topic to the team as we weren't sure of what should be the most convenient.

To keep it short, [they suggested ](https://itducks.slack.com/archives/C01DP38BRTL/p1699279553854939) to keep a short explanation with a link pointing to te full section, instead of directly pointing to the complete page elsewhere.

As nobody else jumped in the thread, and because we were looking for such feedback to make a decision, I chose to stick to that so far, awaiting for more reviews on the thread and the PR. This is a reason why this part has not moved yet.

Using the CLI, you can also deploy a version using the `deploy` command.

```

> bump deploy path/to/file.json --doc my-documentation

Author: 👍 
J'ai modifié l'ordre d'apparition de la note aussi au passage.

```

> You will need to pass your private documentation access token for this command to work. Either with the `--token` flag or via the `BUMP_TOKEN`environment variable. This token can be found from your documentation settings > CI deployment page.
{: .info}

### Deploy with CI

### Deploy using the GitHub Action

Our GitHub Action allows you to easily integrate Bump.sh into your projects by simply adding a workflow file.

If you want to deploy your documentation after each push, you can use the following file: `.github/workflows/bump-deploy.yml`

```
name: Deploy documentation

on:
  push:
    branches:
      - main

jobs:
  deploy-doc:
    name: Deploy API doc on Bump.sh
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Deploy API documentation
        uses: bump-sh/github-action@v1
        with:
          doc: <BUMP_DOC_ID>
          token: ${{secrets.BUMP_TOKEN}}
          file: doc/api-documentation.yml
```

## Release management

Publishing a new version of your documentation typically follows a two-step process:
- You upload a new version of your documentation through a specification file.

> - This new version is validated and deployed, becoming the "visible" version of your documentation. The changelog will also be update, reflecting changes between the current and previous version.

Author: 👍

By default, these two steps are automatically linked: uploading a new version results in its publication. However, it is possible to dissociate them using the Manual Release mode.

The Deployments section of your documentation provides access to the history of all versions uploaded on Bump.sh, including their status (`enqueued`, `deploying`, `deployed`, or `errored`) and their release status (`Released` or `Not released`).

![](/images/help/deployments-list.png)

### Manual Release

> The Manual Release mode was recently introduced on Bump.sh and is still in beta. Please don't hesitate to contact us if you have any questions.

Author: 👍

The Manual Release mode is an option that allows you to separate the upload step from the publication step of your API documentation.

![](/images/help/manual-release-toggle.png)

When activated, uploading a version of your API documentation will no longer automatically publish it. We continue to prepare this version on our side (and you can view a preview and its changelog), but the release step becomes manual. This means you can decide when to make a version live for your API users.

This option allows you to continue working on your documentation without worrying about the impact on your users. For example, you can prepare a future version of your API and its documentation with members of your organization and launch its publication when the time is right.

> #### Release manually a deployment

Reviewer: Isn't this better?
```suggestion
#### Manually release a deployment
```

From the list of your deployments, select a version of your documentation with the "unreleased" status.

![](/images/help/deployments-list-not-released.png)

> The deployment preparation page will appear, allowing you to customize it by adding a title or a brief description, for example, to accompany this new version. These fields support Markdown to make the result more readable.

Reviewer: Add a link to our Markdown support.
It's more about adding context than customizing.

---
title: Deploy and release management
---

- TOC
{:toc}

> ## What are deployments

Author: I was not using the interrogative form here. I feel like keeping it that way.

> Deployment corresponds to the publication of an uploaded API definition of documentation.
> 
> We call an "API document" the file that contains your API definition, the content of your API. When you upload and then deploy an API document, it creates or updates an API version: a marked and named iteration of your API.
> 
> When your API undergoes a structural change, it is reflected on Bump.sh in the changelog. This change is referred to as an "API change."
> 
> ![](/images/help/deployments-list.png)
> 
> By default, an API document is automatically deployed after being uploaded. However, you can choose to deploy it manually at a later time by using the Manual Release mode. In this case, the latest deployments will be enqueued until you decide to release one of them.
> 
> You can also add context (title, description) to any release afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.

Reviewer: I reworked this part to make it easier to understand (and fixed the mistakes). WDYT of this proposal?

```suggestion
A Deployment is the processing of an API document - containing your API definition - sent to Bump.sh servers. 

Following this deployment, the new API definition is published to your API consumers. This is what we call a Release: your API documentation is updated, and if your API has changed since the last deployment, a new entry representing this API change is added to your API changelog.

![](/images/help/deployments-list.png)

By default, a deployment is automatically released after its processing. However, you can choose to control the release flow by activating the Manual Release setting. In this case, the latest deployments will be enqueued until you decide to release one of them.

You can also add context (title, description) to any release afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.
```

(Do not hesitate to fix my English if needed, it's more about simplifying the concepts that working on sentences)

### Deploy from the dashboard

> To deploy a new API definition of documentation from the dashboard, you first need to go to the settings of that documentation.

Reviewer: ```suggestion
To deploy a new API document from the dashboard, you first need to go to the settings of your documentation.
```

> From there, you will find the "Deploy new version" button, which will allow you to choose the file to upload and, if the documentation has branches, the branch to deploy it on.

Reviewer: We need to rename this button!

![](/images/help/upload-document-quick.png)

> After processing, the new API document will be automatically deployed after a few moments unless you have enabled Manual Release mode.

Reviewer: ```suggestion
After processing, the new API document will be automatically released after a few moments unless you have enabled Manual Release mode.
```

### Deploy from the CLI

> Our CLI can also help with deploying an API document using the `deploy` command. The complete process is available on the dedicated [CLI page](/help/continuous-integration/index/).

Reviewer: ```suggestion
You can deploy your API document through our CLI using the `deploy` command. The complete process is available on the dedicated [CLI page](/help/continuous-integration/index/).
```

Reviewer: Also, the link is not the good one (it's the CLI page, not the CI one).

### Deploy using the GitHub Action

Our [GitHub Action](/help/continuous-integration/github-actions/) allows you to easily integrate Bump.sh into your projects by simply adding a workflow file.

## Release management

> Publishing a new API definition of your documentation typically follows a two-step process:

Reviewer: ```suggestion
Releasing a new API definition typically follows a two-step process:
```

> - You upload a new API definition of your documentation through an API document, your specification file.

Reviewer: ```suggestion
- You upload a new API document through your dashboard, the CLI, our GitHub action or our API
```

> - This new API document is validated and deployed, becoming the "visible" API definition of your documentation. The changelog will also be update, reflecting changes between the current and previous definitions.

Reviewer: ```suggestion
- This new API document is validated and deployed, becoming the "visible" API definition presented in your documentation. The changelog is also updated, reflecting changes between the current and previous definitions.
```

> By default, these two steps are automatically linked: uploading a new API document results in its publication. However, it is possible to dissociate them using the Manual Release mode.

Reviewer: ```suggestion
By default, these two steps are automatically linked: uploading a new API document results in its release. However, it is possible to dissociate them using the Manual Release mode.
```

> The Deployments section of your documentation provides access to the history of all API documents uploaded on Bump.sh, including their status (`enqueued`, `deploying`, `deployed`, or `errored`) and their release status (`Released` or `Not released`).

Reviewer: Not sure about this one:
```suggestion
The Deployments section of your documentation provides access to the history of all the API documents uploaded on Bump.sh, including their status (`enqueued`, `deploying`, `deployed`, or `errored`) and their release status (`Released` or `Not released`).
```

![](/images/help/deployments-list.png)

### Manual Release

The Manual Release mode is still in beta. Please don't hesitate to contact us if you have any questions.

> The Manual Release mode is an option that allows you to separate the upload step from the publication step of your API documentation.

Reviewer: ```suggestion
The Manual Release mode is an option that allows you to separate the processing step from the release step of a new API definition.
```

![](/images/help/manual-release-toggle.png)

> When activated, uploading an API document of your API documentation will no longer automatically publish it. We continue to prepare this API definition on our side (and you can view a preview and its changelog), but the release step becomes manual. This means you can decide when to make an API definition live for your API users.

Reviewer: ```suggestion
When activated, uploading an API document will no longer automatically release it. We continue to prepare this API definition on our side (and you can view a preview and its future changelog entry), but the release step becomes manual. This means you can decide when to make an API definition live for your API users.
```

> This option allows you to continue working on your documentation without worrying about the impact on your users. For example, you can prepare a future definition of your API and its documentation with members of your organization and launch its publication when the time is right.

Reviewer: Here is a proposal, to make it a bit simpler to read. WDYT?
```suggestion
This option allows you to work on future versions of your API without disclosing the changes to your users until the moment you choose.
```

#### Manually release a deployment

> From the list of your deployments, select an API document of your documentation with the "unreleased" status.

Reviewer: ```suggestion
Select a deployment with the "unreleased" status from the list of your deployments.
```

![](/images/help/deployments-list-not-released.png)

> The deployment preparation page will appear, allowing you to add context by adding a title or a brief description, for example, to accompany this new definition. These fields [support Markdown](/help/specification-support/markdown-support/) to make the result more readable.

Reviewer: ```suggestion
The release preparation page will appear, allowing you to add context by adding a title or a brief description, for example, to explain the context of the change. These fields [support Markdown](/help/specification-support/markdown-support/) to make the result more readable.
```

![](/images/help/deployment-release-form.png)

> ### Unrelease an API definition

Reviewer: I'm not 100% sure of this one. It's more consistent with the previous title, but it doesn't 100% work. Please ignore it if you think it's better the way it's written now.
```suggestion
### Unrelease a deployment
```

> It is possible to "unrelease" any previously released API definition of your documentation. By clicking on the definition, you will find the "Unrelease" option under the "..." button.

Reviewer: ```suggestion
It is possible to "unrelease" any previously released deployment. By clicking on the deployment, you will find the "Unrelease" option under the "..." button.
```

![](/images/help/unrelease-button.png)

> > Unreleasing a definition automatically adjusts your documentation: it will revert to the previous definition it was using, and its changelog will be adjusted accordingly to correctly reflect the changes.

Reviewer: ```suggestion
> Unreleasing a deployment not only rolls back your API documentation to the previously released one. It also automatically reprocesses its changelog to correctly reflect the changes.
```

{: .info}

> We cannot restore a deleted API document.

> {: .warning}

Reviewer: I don't think we can "delete an API document" at the moment, what are you thinking about here?

Author: Maybe I'm wrong here but can't we delete an API document from the deployment list?
I've tried to remind here that if the user do so, we won't be able to restore the deleted document.

Reviewer 2: Nope, we can only _unrelease_ it. We don't allow suppression so we have a clear history of what has been processed within a documentation.
